### PR TITLE
Classes and Objects & This Pointer

### PR DESCRIPTION
**Pull request by praduman03**
This PR includes the following changes:

1. Added the `userRoutes` file with the User class definition.
2. Using the 'this' pointer in the `userRoutes` file.
3. Created the user API using the above concept.

### CODE REVIEW

**Comment by praduman03** [comment]
I have made an explanation of the concepts [here](https://www.loom.com/share/d4d91418fc4b4946a2940c64b4fc5ab2?sid=bbdb196e-746d-40c1-8ec5-a8210b062c2b).

File: database/config.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const Sequelize = require("sequelize");
+
+const sequelize = new Sequelize("bookabite", "root", "password@03K", {
```
Comment (th3c0d3br34ker) [inline database/config.js]: Please store this configuration in a separate `env` file.

File: routes/userRoutes.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+require("dotenv").config
```
Comment (th3c0d3br34ker) [inline routes/userRoutes.js]: Please do not load the env files again, as you are already loading them in the `index.js` file.

File: routes/userRoutes.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            return res.status(422).json({ message: "Please fill all the required details"})
+        }
+
+        const existingName = await User.findOne({ where: {username}})
+        const existingEmail = await User.findOne({ where: {email}})
+
+        if(existingName || existingEmail){
+            return res.status(409).json({ message: "User with this name or email already exists"})
+        }
+
```
Comment (th3c0d3br34ker) [inline routes/userRoutes.js:55]: Consider adding input sanitization and validation to prevent potential security vulnerabilities.

**Review by th3c0d3br34ker — COMMENTED** [review]
thanks @praduman03, i have reviewed your loom. you have explained everything very clearly.

Overall comments:
1. consider to spilit your `userRouter` file into controllers and services.
2. consider to extend the `Model` class of Sequelize and merge the code of `UserClass` into that class.

i have raised a few concerns from code perspective. Please address those and let me know.

**Comment by praduman03** [comment]
Hey @th3c0d3br34ker, Can I merge now?

**Review by th3c0d3br34ker — APPROVED** [review]
yes, please go ahead and merge this.

**Pull request merged**